Mathlib/FieldTheory/Minpoly/IsConjRoot.lean

```diff
@@ -0,0 +1,310 @@
+  exact fun _ => minpoly.ne_zero h
+
+/--
+`y` is a conjugate root of `x` over `K` if and only if `y` is a root of the minimal polynomial of
+`x`. This is variant of `isConjRoot_iff_aeval_eq_zero`.
+-/
+theorem isConjRoot_iff_mem_minpoly_rootSet {x y : S}
+    (h : IsIntegral K x) : IsConjRoot K x y ↔ y ∈ (minpoly K x).rootSet S :=
+  (isConjRoot_iff_mem_aroots h).trans (by simp [rootSet])
+
+namespace IsConjRoot
+
+/--
+If `y` is a conjugate root of an integral element `x` over `R`, then `y` is also integral
+over `R`.
+-/
+theorem isIntegral {x y : A} (hx : IsIntegral R x) (h : IsConjRoot R x y) :
+    IsIntegral R y :=
+  ⟨minpoly R x, minpoly.monic hx, h ▸ minpoly.aeval R y⟩
+
+/--
+A variant of `IsConjRoot.eq_of_isConjRoot_algebraMap`, only assuming `Nontrivial R`,
+`NoZeroSMulDivisors R A` and `Function.Injective (algebraMap R A)` instead of `Field R`. If `x` is a
+conjugate root of some element `algebraMap R S r` in the image of the base ring, then
+`x = algebraMap R S r`.
+-/
+theorem eq_algebraMap' [Nontrivial R] [NoZeroSMulDivisors R S] {r : R} {x : S}
+    (h : IsConjRoot R x (algebraMap R S r)) (hf : Function.Injective (algebraMap R S)) :
+    x = algebraMap R S r := by
+  rw [IsConjRoot, minpoly.eq_X_sub_C_of_algebraMap_inj _ hf] at h
+  have : x ∈ (X - C r).aroots S := by
+    rw [mem_aroots]
+    simp [X_sub_C_ne_zero, h ▸ minpoly.aeval R x]
+  simpa [aroots_X_sub_C] using this
+
+/--
+If `x` is a conjugate root of some element `algebraMap R S r` in the image of the base ring, then
+`x = algebraMap R S r`.
+-/
+theorem eq_algebraMap {r : K} {x : S} (h : IsConjRoot K x (algebraMap K S r)) :
+    x = algebraMap K S r :=
+  eq_algebraMap' h (algebraMap K S).injective
+
+/--
+A variant of `IsConjRoot.eq_zero`, only assuming `Nontrivial R`,
+`NoZeroSMulDivisors R A` and `Function.Injective (algebraMap R A)` instead of `Field R`. If `x` is a
+conjugate root of `0`, then `x = 0`.
+-/
+theorem eq_zero' [Nontrivial R] [NoZeroSMulDivisors R S] {x : S} (h : IsConjRoot R 0 x)
+    (hf : Function.Injective (algebraMap R S)) : x = 0 :=
+  (algebraMap R S).map_zero ▸ (eq_algebraMap' ((algebraMap R S).map_zero ▸ h.symm) hf)
+
+/--
+If `x` is a conjugate root of `0`, then `x = 0`.
+-/
+theorem eq_zero {x : S} (h : IsConjRoot K 0 x) : x = 0 :=
+  eq_zero' h (algebraMap K S).injective
+
+end IsConjRoot
+
+/--
+A variant of `IsConjRoot.iff_eq_zero'`, only assuming `Nontrivial R`,
+`NoZeroSMulDivisors R A` and `Function.Injective (algebraMap R A)` instead of `Field R`. `x` is a
+conjugate root of `0` if and only if `x = 0`.
+-/
+theorem isConjRoot_zero_iff_eq_zero' [Nontrivial R] {x : S} [NoZeroSMulDivisors R S]
+    (hf : Function.Injective (algebraMap R S)) : IsConjRoot R 0 x ↔ x = 0 :=
+  ⟨fun h => eq_zero' h hf, fun h => h.symm ▸ rfl⟩
+
+/--
+`x` is a conjugate root of `0` if and only if `x = 0`.
+-/
+@[simp]
+theorem isConjRoot_zero_iff_eq_zero {x : S} : IsConjRoot K 0 x ↔ x = 0 :=
+  isConjRoot_zero_iff_eq_zero' (algebraMap K S).injective
```

Comment:
Could you also add the symmetric version as `@[simp]` lemma, stating `IsConjRoot K x 0 ↔ x = 0`?

Reply:
Thank you! Added.